vochain/indexer/indexer.go

```diff
@@ -727,3 +737,78 @@ func (idx *Indexer) GetTokenFeesByType(txType string, offset, maxItems int32) ([
 	}
 	return tt, nil
 }
+
+// GetTokenTransfersByToAccount returns all the token transfers made to a given account
+// from the database, ordered by timestamp and paginated by maxItems and offset
+func (idx *Indexer) GetTokenTransfersByToAccount(to []byte, offset, maxItems int32) ([]*indexertypes.TokenTransferMeta, error) {
+	ttFromDB, err := idx.readOnlyQuery.GetTokenTransfersByToAccount(context.TODO(), indexerdb.GetTokenTransfersByToAccountParams{
+		ToAccount: to,
+		Limit:     int64(maxItems),
+		Offset:    int64(offset),
+	})
+	if err != nil {
+		return nil, err
+	}
+	tt := []*indexertypes.TokenTransferMeta{}
+	for _, t := range ttFromDB {
+		tt = append(tt, &indexertypes.TokenTransferMeta{
+			Amount:    uint64(t.Amount),
+			From:      t.FromAccount,
+			To:        t.ToAccount,
+			Height:    uint64(t.BlockHeight),
+			TxHash:    t.TxHash,
+			Timestamp: t.TransferTime,
+		})
+	}
+	return tt, nil
+}
+
+// GetTokenTransfersByAccount returns all the token transfers made to and from a given account
+// from the database, ordered by timestamp and paginated by maxItems and offset
+func (idx *Indexer) GetTokenTransfersByAccount(acc []byte, offset, maxItems int32) (map[string][]*indexertypes.TokenTransferMeta, error) {
+	transfersTo, err := idx.GetTokenTransfersByToAccount(acc, offset, maxItems)
+	if err != nil {
+		return nil, err
+	}
+	transfersFrom, err := idx.GetTokenTransfersByFromAccount(acc, offset, maxItems)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string][]*indexertypes.TokenTransferMeta{
+		"Received": transfersTo,
+		"Sent":     transfersFrom,
+	}, nil
```

Comment:
personally I wouldn't use a map with always the same two string keys - a struct will work much better